### PR TITLE
Feature/dim user proficiencies

### PR DESCRIPTION
**Pull request by allison-code-dot-org**
# Description

This PR creates the model called dim_l3_proficiencies, which includes all students who reach "L3 proficiency" in CSF levels. This model is meant to be aggregated at both the calendar and school year levels. This model will only be used by the global team until new metrics are released in the "learning measures" project. 

## Links

Jira ticket(s): [DATAOPS-1064](https://codedotorg.atlassian.net/jira/software/projects/DATAOPS/boards/72/backlog?selectedIssue=DATAOPS-1064&text=prof)

### CODE REVIEW

File: dbt/models/marts/students/_students__models.yml [hunk 1]
```diff
@@ -260,5 +260,31 @@ models:
       - name: time_spent_minutes 
         description: the number of minutes the student spent on that level on the particular date
     
+    config:
+      tags: ['released']
+  
+  - name: dim_l3_proficiencies
+    description: |
+      This model has a row for every time a student reaches L3 proficiency within CSF levels. This proficiency threshold is set by our curriculum team per script. There may be multiple rows per student, since they can reach L3 proficiency multiple times.
```
Comment (jordan-springer) [inline dbt/models/marts/students/_students__models.yml:268]: if you want to know if a student reached l3_proficiency, we should take the first instance of them reaching that level (eg. min(basic_proficiency_at)

File: dbt/models/marts/students/dim_l3_proficiencies.sql [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    extract(year from basic_proficiency_at)     as cal_year,
+    school_years.school_year                    as school_year,
+    students.country                            as country,
+    basic_proficiency_at                        as l3_proficiency_at
+from user_proficiencies
+join students
+    on user_proficiencies.user_id = students.student_id
+join school_years 
+    on user_proficiencies.basic_proficiency_at between school_years.started_at and school_years.ended_at
+where basic_proficiency_at is not null
```
Comment (jordan-springer) [inline dbt/models/marts/students/dim_l3_proficiencies.sql:29]: move to subquery

File: dbt/models/marts/students/dim_l3_proficiencies.sql [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    select * 
+    from {{ ref('dim_students') }}
+),
+
+school_years as (
+    select * 
+    from {{ ref('int_school_years') }}
+)
+
+select distinct
```
Comment (jordan-springer) [inline dbt/models/marts/students/dim_l3_proficiencies.sql:18]: distinct is going to flatten your multiple student rows, instead use the min(basic_proficiency_at) to show when it was first achieved and group by other dimensions

File: dbt/models/marts/students/dim_l3_proficiencies.sql [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    from {{ ref('dim_students') }}
+),
+
+school_years as (
+    select * 
+    from {{ ref('int_school_years') }}
+)
+
+select distinct 
+    user_id                                     as student_id,
```
Comment (jordan-springer) [inline dbt/models/marts/students/dim_l3_proficiencies.sql:19]: alias columns

**Review by jordan-springer — CHANGES_REQUESTED** [review]
remove distinct and replace with agg on when proficiency was reached

File: dbt/models/marts/students/_students__models.yml (hunk 1 above)
Comment (allison-code-dot-org) [reply dbt/models/marts/students/_students__models.yml:268]: this is how I originally had it, but we need the ability to group by both the calendar year and the school year. So, taking the min with those two variables did not make sense.

**Pull request merged**